internal/controller/drplacementcontrol.go

```diff
@@ -1549,12 +1549,51 @@ func (d *DRPCInstance) ensureVRGManifestWork(homeCluster string) error {
 	d.log.Info("Ensure VRG ManifestWork",
 		"Last State:", d.getLastDRState(), "cluster", homeCluster)
 
-	cachedVrg := d.vrgs[homeCluster]
-	if cachedVrg == nil {
-		return fmt.Errorf("failed to get vrg from cluster %s", homeCluster)
+	mw, mwErr := d.mwu.FindManifestWorkByType(rmnutil.MWTypeVRG, homeCluster)
+	if mwErr != nil {
+		if errors.IsNotFound(mwErr) {
+			cachedVrg := d.vrgs[homeCluster]
+			if cachedVrg == nil {
+				return fmt.Errorf("failed to get vrg from cluster %s", homeCluster)
+			}
+
+			return d.createVRGManifestWork(homeCluster, cachedVrg.Spec.ReplicationState)
+		}
+
+		return fmt.Errorf("ensure VRG ManifestWork failed (%w)", mwErr)
+	}
+
+	vrg, err := rmnutil.ExtractVRGFromManifestWork(mw)
+	if err != nil {
+		return fmt.Errorf("error extracting VRG from ManifestWork for cluster %s. Error: %w", homeCluster, err)
+	}
+
+	d.updateVRGOptionalFields(vrg, homeCluster)
+
+	return d.mwu.AddVRGToManifestWork(vrg, mw)
+}
+
+// updateVRGOptionalFields ensures that the VRG object optional fields are up to date.
+// The fields that this function will NOT update are:
+//   - ObjectMeta.Name
+//   - ObjectMeta.Namespace
+//   - Spec.PVCSelector
+//   - Spec.ReplicationState
+//   - Spec.VolSync.RDSpec
```

Comment:
(nit) These fields are also not updated and should be added here (for later reading :) )
- PrepareForFinalSync
- RunFinalSync

If volsync was disabled and later enabled, do we need to flip spec.volsync.disabled here? Or should we ignore it this update?

Reply:
PrepareForFinalSync and RunFinalSync Added to the comment. 
Good catch for `spec.volsync.disabled`. It should be consolidated within this function.